module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -966,19 +966,15 @@ protected function getPagedResults($responseKey, $interface, $query = [])
                 $msg = $json->errors[0]->message ?? json_last_error_msg();
                 throw new ILSException("Error: '$msg' fetching '$responseKey'");
             }
-            $total = $json->totalRecords ?? 0;
-            $previousCount = $count;
+            $count = 0;
             foreach ($json->$responseKey ?? [] as $item) {
                 $count++;
-                if ($count % $limit == 0) {
-                    $offset += $limit;
-                }
                 yield $item ?? '';
             }
-            // Continue until the count reaches the total records
-            // found, if count does not increase, something has gone
-            // wrong. Stop so we don't loop forever.
-        } while ($count < $total && $previousCount != $count);
+            $offset += $limit;
+
+            // Continue until the results are not the limit value (i.e. the last page of results)
+        } while ($count == $limit);
```

Comment:
That is a good idea. We've seen that scenario occur with FOLIO's OAI harvest calls, so it's not out of the realm of possibility that other API calls could have that happen at some point. I made a commit with that suggestion in it.

Reply:
That sounds good. I just double checked in our FOLIO instance how the `totalRecords` field would behave when thousands of records were returned to make sure that the value wasn't always the low ~1000 value like it was on the `offset=0` pass, because I couldn't recall, and it does increase like we would need it to so I'll go ahead and work on a commit for this change.

| limit | offset | totalRecords |
|-------|--------|---------------|
| 1000  | 0      | 1000          |
| 1000  | 1000   | 2000          |
| 1000  | 2000   | 3000          |
| 1000  | 3000   | 3042          |